### DISCUSSION TITLE
[Auth0 Client] signup

### DISCUSSION THREAD

**Discussion by georgev-97** [mia-platform/community#362] [Q&A] [answered]
### Which section?

Console

### Which is the version?

12

### Description

Hello guys,
as stated in the [documentation](https://docs.mia-platform.eu/docs/runtime_suite/auth0-client/overview_and_usage) the micro-service exposes the following routes:  `/authorize, /oauth/token, /refreshtoken, /logout, /userinfo and /users/me.`

The auth0 IDP provides two different signup methods, one via the authentication API `POST /dbconnections/signup` and one by using the user management service `POST  /api/v2/users`.
Can I access those endpoints via the auth0 client or should I access them directly by contacting the IDP?

**Comment by malta895** [comment, marked as answer]
Hello @georgev-97,
Just for clarification, what is your use case? Aren't you able to use [Auth0 Universal Login](https://auth0.com/docs/authenticate/login/auth0-universal-login)?

In case you are not, the Auth0 client offers the possibility to manage the users via a dedicated set of APIs, in particular you can use the API `POST /users`.

Be aware that this is a User Management API, for this reason it needs to be called by the back-end of your application, which in turn will expose a signup endpoint to the client, that you need to implement.

You can contact this API with the same body accepted by Auth0 [POST /api/v2/users](https://auth0.com/docs/api/management/v2/users/post-users) API, with the only difference that the `connection` parameter is optional: if omitted will default on the `managementClient.defaultCreateUserConnection` configuration parameter.
Refer to [the documentation](https://docs.mia-platform.eu/docs/runtime_suite/auth0-client/configuration#configuration) for a complete configuration schema.

Be also aware that when you use the Auth0-client service, all the authentication management should be performed through it, and not by calling Auth0 APIs directly, otherwise sessions won't be recognized.

**Reply by georgev-97** [reply]
I'm not able to use the Universal Login because I have to use internal signin/signup pages.
Anyway the `POST /users` will work for my use case, thank you!

**Reply by malta895** [reply]
You're welcome!
